classpath/java/util/ArrayList.java

```diff
@@ -183,6 +183,20 @@ public void clear() {
   public ListIterator<T> listIterator() {
     return listIterator(0);
   }
+  
+  @Override
+  public boolean retainAll(Collection<?> c) {
+    boolean changed = false;
+    Iterator<T> iter = this.iterator();
+    while(iter.hasNext()) {
+      T item = iter.next();
+      if(!c.contains(item)) {
+        changed = true;
+        iter.remove();
```

Comment:
Should we be removing the iterator when still in a loop? If changed is true, would it be good to return changed from there and break from the loop?